astropy/modeling/functional_models.py

```diff
@@ -400,13 +400,15 @@ def evaluate(x, factor):
         return factor * x
 
 
-class Redshift(Fittable1DModel):
+class RedshiftWavelength(Fittable1DModel):
     """
     One dimensional redshift model.
 
+    Redshifts a wavelength array from rest (z=0) to the input redshift.
+
     Parameters
     ----------
-    z : float or a list of floats
+    z : int or float
```

Comment:
It's just a general feature of models.  Whether or not it's useful for a specific model or usecase is up to the user to decide.